internal/gps/solve_basic_test.go

```diff
@@ -1550,6 +1553,72 @@ func (sm *depspecSourceManager) ignore() map[string]bool {
 	return sm.ig
 }
 
+// DeduceConstraint tries to puzzle out what kind of version is given in a string -
+// semver, a revision, or as a fallback, a plain tag
+func (sm *depspecSourceManager) DeduceConstraint(s string, pi ProjectIdentifier) (Constraint, error) {
```

Comment:
This probably isn't the place to put a real implementation. The `depspecSourceManager` is specifically, and exclusively (at least for now) used to provide a `SourceManager` for solver tests, based on an input depspec. These depspecs do generally rely on semver, but they don't use "real" revisions or anything - just arbitrary strings, as version identifiers aren't the system under test.

In fact, the implementation here should probably be a panic, as there's no current circumstance under which the `depspecSourceManager` is useful outside of the gps solving tests, and it shouldn't be used anywhere else without a conscious and intentional expansion of its semantics. A panic would prevent inadvertent usage and scope creep.